src/content/docs/node/introduction.mdx

```diff
@@ -1,3 +1,502 @@
+function add(a, b) {
+  return a + b;
+}
+
+// Export the function
+module.exports = { add };
+```
+
+Here, the add function is exported using module.exports, which makes it available to other files.
+
+**Importing a Module**
+Once a module is created and exported, it can be imported and used in other files using the require function:
+
+```js
+//app.js
+
+const math = require("./math");
+console.log(math.add(2, 3)); // Output: 5
+```
+
+### CommonJS vs. ES Modules
+
+Node.js supports two module systems: **CommonJS** (the older, default module system) and **ES Modules** (a more modern, standards-based approach). Let's compare their differences:
+
+#### Syntax Comparison
+
+CommonJS uses `module.exports` and `require` for exporting and importing modules, while ES Modules use `export` and `import`.
+
+| Feature        | CommonJS                          | ES Modules                                                  |
+| -------------- | --------------------------------- | ----------------------------------------------------------- |
+| Syntax         | `module.exports` / `require`      | `export` / `import`                                         |
+| File Extension | `.js`                             | `.mjs` (or `.js` with `"type": "module"` in `package.json`) |
+| Import Type    | Synchronous                       | Asynchronous                                                |
+| Default Usage  | Older Node.js apps, simpler setup | Modern apps, full browser support                           |
+
+##### Example: CommonJS
+
+```js
+// math.js
+module.exports = { add: (a, b) => a + b };
+
+// app.js
+const math = require("./math");
+console.log(math.add(2, 3));
+```
+
+##### Example: ES Modules
+
+```js
+// math.mjs
+export function add(a, b) {
+  return a + b;
+}
+
+// app.mjs
+import { add } from "./math.mjs";
+console.log(add(2, 3));
+```
+
+### Built-in Modules in Node.js
+
+Node.js provides several built-in modules that are essential for building applications. These modules cover common tasks like file handling, networking, and event handling. Here are a few important ones:
+
+##### 1. File System (`fs`)
+
+The `fs` module allows you to interact with the file system, enabling you to read, write, update, and delete files.
+
+```js
+// Import the 'fs' module
+const fs = require("fs");
+
+fs.readFile("example.txt", "utf8", (err, data) => {
+  if (err) throw err;
+  console.log(data);
+});
+```
+
+##### 1. File System (`fs`)
+
+The `fs` module allows you to interact with the file system, enabling you to read, write, update, and delete files.
+
+**Example: Reading a file**
+
+```js
+// Import the 'fs' module
+const fs = require("fs");
+
+fs.readFile("example.txt", "utf8", (err, data) => {
+  if (err) throw err;
+  console.log(data);
+});
+```
+
+##### 2. HTTP
+
+The `http` module enables the creation of web servers and handling of HTTP requests.
+**Example: Simple HTTP server**
+
+```js
+const http = require("http");
+
+http
```

Comment:
Seems like there is an error in this code snippet please re verify this code snippet and update it accordingly